graph/appender/appender.go

```diff
@@ -1,14 +1,75 @@
 package appender
 
 import (
+	"github.com/kiali/kiali/business"
+	"github.com/kiali/kiali/config"
 	"github.com/kiali/kiali/graph"
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/models"
+	"github.com/kiali/kiali/prometheus"
 )
 
+// GlobalInfo caches information relevant to a single graph. It allows
+// an appender to populate the cache and then it, or another appender
+// can re-use the information.  A new instance is generated for graph and
+// is initially empty.
+type GlobalInfo struct {
+	Business    *business.Layer
+	IstioClient *kubernetes.IstioClient
+	PromClient  *prometheus.Client
+}
+
+func NewGlobalInfo() *GlobalInfo {
+	return &GlobalInfo{}
+}
+
+// NamespaceInfo caches information relevant to a single namespace. It allows
+// one appender to populate the cache and another to then re-use the information.
+// A new instance is generated for each namespace of a single graph and is initially
+// seeded with only Namespace.
+type NamespaceInfo struct {
+	Namespace        string // always provided
+	ExternalServices map[string]bool
+	WorkloadList     *models.WorkloadList
+}
+
+func NewNamespaceInfo(namespace string) *NamespaceInfo {
+	return &NamespaceInfo{Namespace: namespace}
+}
+
+func getWorkload(workloadName string, workloadList *models.WorkloadList) *models.WorkloadListItem {
+	for _, workload := range workloadList.Workloads {
+		if workload.Name == workloadName {
+			return &workload
+		}
+	}
+	return nil
+}
+
+func getAppWorkloads(app, version string, workloadList *models.WorkloadList) []*models.WorkloadListItem {
```

Comment:
There is a GetAppList / GetApp doing a similar logic, I wonder if we can use that, instead to build the same logic (i.e. fetch workloads for an app) in several places.
Also, we can modify the GetAppList / GetApp is something needs to be adjusted, but I prefer to locate logic in a single place for easy change.

Reply:
@lucasponce ,Note that this is not fetching, it is narrowing a list previously fetched from the business layer.  It's just a little utility function.